mypy/graph_utils.py

```diff
@@ -57,7 +57,11 @@ def prepare_sccs(
     sccs: list[set[T]], edges: dict[T, list[T]]
 ) -> dict[AbstractSet[T], set[AbstractSet[T]]]:
     """Use original edges to organize SCCs in a graph by dependencies between them."""
-    sccsmap = {v: frozenset(scc) for scc in sccs for v in scc}
```

Comment:
If you wanted to keep the dict comprehension, you could use an assignment expression
```py
sccsmap = {v: s for scc in sccs if (s := frozenset(scc)) is not None for v in scc}  # type: ignore[redundant-expr]
```

It adds an additional `if not None` check for each `scc` that's always true but necessary to use `:=`.